test/successful-files.test.js

```diff
@@ -43,3 +43,45 @@ test('defined YAML config path', makeFileTest('fixtures/foo.yaml'));
 test('defined JS config path', makeFileTest('fixtures/foo.js'));
 
 test('defined modulized JS config path', makeFileTest('fixtures/foo-module.js'));
+
+test('transform', function (assert) {
+  function transform(result) {
+    result.config.foo = [result.config.foo];
+    return result;
+  }
+
+  // for testing transform, it should be enough to check for any 1 file type
+  var filePath = absolutePath('fixtures/foo.json');
+  var loadConfig = cosmiconfig(null, {
+    transform: transform,
+  }).load;
+  var loadConfigSync = cosmiconfig(null, {
+    sync: true,
+    transform: transform,
+  }).load;
+
+  try {
+    var result = loadConfigSync(null, filePath);
+
+    assert.deepEqual(
+      result.config,
+      { foo: [true] },
+      'Result config should be transformed in sync'
+    );
+
+    loadConfig(null, filePath).then(function (result) {
+
+      assert.deepEqual(
+        result.config,
+        { foo: [true] },
+        'Result config should be transformed in async'
+      );
+
+      assert.end();
+    }).catch(function (err) {
+      assert.end(err);
+    });
+  } catch (err) {
+    assert.end(err);
+  }
+});
```

Comment:
I was thinking about doing that for this one because it's new, so easy to make clear. The others from your sync PR are fine for now, I think.